x-pack/legacy/plugins/maps/public/connected_components/map/mb/utils.js

```diff
@@ -142,3 +144,11 @@ export async function addSpritesheetToMap(json, imgUrl, mbMap) {
   const imgData = await loadSpriteSheetImageData(imgUrl);
   addSpriteSheetToMapFromImageData(json, imgData, mbMap);
 }
+
+function isCrossOriginUrl(url) {
+  const a = window.document.createElement('a');
+  a.href = url;
+  return (
+    a.protocol !== window.document.location.protocol || a.host !== window.document.location.host
```

Comment:
There's a small corner case that this code doesn't cover. [Two origins are only the same](https://tools.ietf.org/html/rfc6454#section-5) iff they have identical schemes, hosts, _and_ ports.

```suggestion
    a.protocol !== window.document.location.protocol || a.host !== window.document.location.host || a.port !== window.document.location.port
```
I would suggest comparing the [Location origin Property](https://www.w3schools.com/jsref/prop_loc_origin.asp) with the [Anchor origin Property](https://www.w3schools.com/jsref/prop_anchor_origin.asp), but the latter isn't supported in IE 😢

See also: [`parse_next.ts`](../../blob/master/x-pack/legacy/plugins/security/public/lib/parse_next.ts) -- this isn't directly applicable here because you're working purely with the DOM, but there's an extra "gotcha" when using the Node URL parser.